lib/mongo/caching_cursor.rb

```diff
@@ -0,0 +1,75 @@
+# Copyright (C) 2020 MongoDB Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+module Mongo
+
+  # A Cursor that attempts to load documents from memory first before hitting
+  # the database if the same query has already been executed.
+  #
+  # @api semiprivate
+  class CachingCursor < Cursor
+
+    # @return [ Array <BSON::Document> ] The cursor's cached documents.
+    # @api private
+    attr_reader :cached_docs
+
+    # @return [ Integer ] The max number of docs to return from the query.
+    attr_accessor :limit_docs
```

Comment:
Correct me if I'm wrong here -- it seems like this is the limit on the view? If so, you can eliminate this method because the `Cursor` class already has a `limit` method. Let me know if I'm missing something and there's some reason you can't use that method.